procrastinate/structure.sql

```diff
@@ -79,22 +96,82 @@ BEGIN
 END;
 $$;
 
-CREATE SEQUENCE public.procrastinate_jobs_id_seq;
-
-ALTER SEQUENCE public.procrastinate_jobs_id_seq OWNED BY public.procrastinate_jobs.id;
+CREATE FUNCTION procrastinate_trigger_status_events_procedure_insert() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    INSERT INTO procrastinate_events(job_id, type)
+        VALUES (NEW.id, 'deferred'::procrastinate_job_event_type);
+	RETURN NEW;
+END;
+$$;
 
-ALTER TABLE ONLY public.procrastinate_jobs ALTER COLUMN id
-    SET DEFAULT nextval('public.procrastinate_jobs_id_seq'::regclass);
+CREATE FUNCTION procrastinate_trigger_status_events_procedure_update() RETURNS trigger
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    WITH t AS (
+        SELECT CASE
+            WHEN OLD.status = 'todo'::procrastinate_job_status
+                AND NEW.status = 'doing'::procrastinate_job_status
+                THEN 'started'::procrastinate_job_event_type
+            WHEN OLD.status = 'doing'::procrastinate_job_status
+                AND NEW.status = 'todo'::procrastinate_job_status
+                THEN 'retried'::procrastinate_job_event_type
```

Comment:
Maybe `retried` should be `to_be_retried`?

Reply:
`deferred_retry` ?

Reply:
`deferred_for_retry` sounds :ok_hand: